akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/ClusterSharding.scala

```diff
@@ -231,6 +243,58 @@ object ShardedEntity {
     typeKey:        EntityTypeKey[M],
     stopMessage:    M): ShardedEntity[M, ShardingEnvelope[M]] =
     new ShardedEntity(createBehavior, typeKey, stopMessage, Props.empty, Optional.empty(), Optional.empty(), Optional.empty())
+
+  // FIXME remove above create in favor of below `of`
+
+  /**
+   * Defines how the entity should be created. Used in [[ClusterSharding#start]]. More optional
+   * settings can be defined using the `with` methods of the returned [[ShardedEntity]].
+   *
+   * @param createBehavior Create the behavior for an entity given an entityId
+   * @param typeKey A key that uniquely identifies the type of entity in this cluster
+   * @param stopMessage Message sent to an entity to tell it to stop, e.g. when rebalanced or passivated.
+   *
+   * @tparam M The type of message the entity accepts
+   */
+  def of[M](
+    createBehavior: JFunction[String, Behavior[M]],
+    typeKey:        EntityTypeKey[M],
+    stopMessage:    M): ShardedEntity[M, ShardingEnvelope[M]] = {
+    create(new BiFunction[ActorRef[ClusterSharding.ShardCommand], String, Behavior[M]] {
+      override def apply(shard: ActorRef[ClusterSharding.ShardCommand], entityId: String): Behavior[M] =
+        createBehavior.apply(entityId)
+    }, typeKey, stopMessage)
+  }
+
+  /**
+   * Defines how the [[PersistentEntity2]] should be created. Used in [[ClusterSharding#start]]. More optional
+   * settings can be defined using the with` methods of the returned [[ShardedEntity]].
+   *
+   * @param createBehavior Create the behavior for an entity given an entityId
+   * @param typeKey A key that uniquely identifies the type of entity in this cluster
+   * @param stopMessage Message sent to an entity to tell it to stop, e.g. when rebalanced or passivated.
+   *
+   * @tparam Command The type of message the entity accepts
+   */
+  def ofPersistentEntity[Command, Event, State >: Null](
+    createPersistentEntity: JFunction[String, PersistentEntity2[Command, Event, State]],
```

Comment:
I did a small refinement here. Instead of having two overloaded `start` methods I added this factory method for `ShardedEntity`. See usage in [PersistentEntity3CompileOnlyTest](https://github.com/akka/akka/pull/25697/files#diff-0b7d0e19bcd9431650320ad44d241a74).

This should be equally discoverable and then we don't have to have a different two variants of ShardedEntity that are exactly the same aside from this factory.

Reply:
Perhaps `ofPersistent` since the non-persistent one is merely called `of`?

Reply:
I like it. But, actually, I was rather thinking that `ofEntity` and of `ofPersistentEntity` is more clear and guided.

Reply:
I think I would go with "entityOf" and "persistentEntityOf".

Reply:
We should read this as:
```
ShardedEntity.of(...)
ShardedEntity.ofPersistentEntity(...)
```

I like @viktorklang 's proposal:
```
ShardedEntity.of(...)
ShardedEntity.ofPersistent(...)
```

@jonas 's suggestion also works for me although it's a bit of repetition.

another one could be:
```
ShardedEntity.ofBehavior(...)
ShardedEntity.ofPersistentEntity(...)
```

@WadeWaldron 's :
```
ShardedEntity.entityOf(...)
ShardedEntity.persistentEntityOf(...)
```
while that might read nice once it's written it's more difficult to see that these are two alternatives to choose between when using code completion in IDE. I like starting such with same prefix. Another reason for using `of` is that it's a pretty well know naming for factor methods in Java.

PS. Note that the `.of` method also accepts PersistentEntity since PersistentEntity __is a__ Behavior. The only purpose for having both is to give the user a better hint of the existence of PersistentEntity, which was what you asked for as I understand it.

Reply:
I think they need to start with the same word so that they are adjacent in an auto-complete/intellisense situation. Therefor I prefer "of" and "ofPersistent"

Reply:
I agree that starting with `of` is important.

Reply:
Also agree with staring with `of` for discoverability, but also because `..Entity.enity..` is kind of repetitive. 

For alignment, we should use either the term `Entity` or `Behavior`, but not mixing them. 

```scala
ShardedEntity.ofBehavior(...)
ShardedEntity.ofPersistentBehavior(...)
```

```scala
ShardedEntity.ofEntity(...)
ShardedEntity.ofPersistentEntity(...)
```

I personally think that it should be `Behavior`. We are creating an `EntityRef` from a `Behavior`.

Reply:
Fair argument. I can go with `of...` for discoverability.